### PR TITLE
cratification: Example support

### PR DESCRIPTION
**Pull request by stormasm**
# Description

When the crate nu_cmd_lang crate was created last week example_test.rs was copied over from nu_command
to nu_cmd_lang.  By doing this there was a set of methods in example_test.rs that existed in both crates...

This PR removes the redundancy by moving all of those duplicated methods into the crate nu_test_support in a newly created file called example_support.rs

_(Thank you for improving Nushell. Please, check our [contributing guide](../CONTRIBUTING.md) and talk to the core team before making major changes.)_

_(Description of your pull request goes here. **Provide examples and/or screenshots** if your changes affect the user experience.)_

# User-Facing Changes

_(List of all changes that impact the user experience here. This helps us keep track of breaking changes.)_

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-cmd-lang/src/example_test.rs [hunk 1]
```diff
@@ -6,15 +6,19 @@ pub fn test_examples(cmd: impl Command + 'static) {
     test_examples::test_examples(cmd);
 }
 
+#[cfg(test)]
```
Comment (jntrnr) [inline crates/nu-cmd-lang/src/example_test.rs:9]: accidental duplication?

**Review by sholderbach — COMMENTED** [review]
Thanks for cutting that down!

I have a potential concern about our dependency tree. Maybe we need something like `nu-cmd-lib` to keep common stuff, that needs to depend on the foundational nu crates, only.

File: Cargo.lock [hunk 2]
```diff
@@ -2950,8 +2950,12 @@ version = "0.76.1"
 dependencies = [
  "getset",
  "hamcrest2",
+ "itertools",
+ "nu-engine",
  "nu-glob",
+ "nu-parser",
  "nu-path",
+ "nu-protocol",
```
[the remark this answers is not in the document]
Comment (stormasm) [reply Cargo.lock]: @sholderbach agreed, I dramatically cleaned up this PR and removed all of that dependency head ache...
Now its way cleaner and much more simple...

File: crates/nu-cmd-lang/src/example_test.rs (hunk 1 above)
Comment (stormasm) [reply crates/nu-cmd-lang/src/example_test.rs:9]: good catch ! thanks....  its now gone...

**Review by sholderbach — APPROVED** [review]
Thanks for the quick turnaround! Looking forward to the further steps!

**Pull request merged**